### PR TITLE
bpo-46430: Add test for checking block leak is zero

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue46430

### CODE REVIEW

File: Lib/test/test_cmd_line.py [hunk 1]
```diff
@@ -125,6 +126,9 @@ def run_python(*args):
             # because of bugs in C extensions. This test is only about checking
             # the showrefcount feature.
             self.assertRegex(err, br'^\[-?\d+ refs, \d+ blocks\]')
+            refs, blocks = map(int, re.findall(r'-?\d+', str(err)))
+            self.assertLessEqual(refs, 0)
```
Comment (kumaraditya303) [inline Lib/test/test_cmd_line.py:130]: Refs should be 0 not negative.

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: Lib/test/test_cmd_line.py (hunk 1 above)
Comment (corona10) [reply Lib/test/test_cmd_line.py:130]: @kumaraditya303 Please read the comment why I check the negotive value

`````

            # bpo-46417: Tolerate negative reference count which can occur
            # because of bugs in C extensions. This test is only about checking
            # the showrefcount feature.
`````

Comment (corona10) [reply Lib/test/test_cmd_line.py:130]: But I am closing this PR since the https://github.com/python/cpython/pull/31555  looks better approach.
